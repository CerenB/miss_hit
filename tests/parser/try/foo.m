% From https://uk.mathworks.com/help/matlab/ref/try.html

function foo()
    x = 5;
    y = 0;
    try
        z = x / y;
    catch
        z = 1;
    end
end
