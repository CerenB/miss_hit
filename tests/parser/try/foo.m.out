----------------------------------------------------------------------
--  Parse tree for foo.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <foo>
    Body: Sequence_Of_Statements
      Statements: Simple_Assignment_Statement
        LHS: Identifier <x>
        RHS: Number_Literal <5>
      Statements: Simple_Assignment_Statement
        LHS: Identifier <y>
        RHS: Number_Literal <0>
      Statements: Try_Statement
        Body: Sequence_Of_Statements
          Statements: Simple_Assignment_Statement
            LHS: Identifier <z>
            RHS: Binary_Operation /
              LHS: Identifier <x>
              RHS: Identifier <y>
        Handler: Sequence_Of_Statements
          Statements: Simple_Assignment_Statement
            LHS: Identifier <z>
            RHS: Number_Literal <1>
----------------------------------------------------------------------
In foo.m, line 3
| function foo()
|             ^ info: this token is not linked to the ast
In foo.m, line 3
| function foo()
|              ^ info: this token is not linked to the ast
In foo.m, line 4
|     x = 5;
|          ^ info: this token is not linked to the ast
In foo.m, line 5
|     y = 0;
|          ^ info: this token is not linked to the ast
In foo.m, line 7
|         z = x / y;
|                  ^ info: this token is not linked to the ast
In foo.m, line 9
|         z = 1;
|              ^ info: this token is not linked to the ast
In foo.m, line 10
|     end
|     ^^^ info: this token is not linked to the ast
In foo.m, line 11
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
